File: server/makc2022--nodejs--nestjs/src/layer1/query/query-resource.ts
/** Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License. */

import { I18nContext } from 'nestjs-i18n';
import { IQueryResource } from './query-resource.interface';

/** Ресурс запроса. */
export class QueryResource implements IQueryResource {
  private path = 'layer1--query';

  /** Конструктор.
   * @param i18n Контекст интернационализации.
   */
  constructor(private readonly i18n: I18nContext) {}

  /** @inheritdoc */
  getErrorMessageForDefault(): string {
    return this.i18n.t(`${this.path}.@@ErrorMessageForDefault`);
  }

  /** @inheritdoc */
  getErrorMessageForInvalidInput(invalidProperties: string[]): string {
    return this.i18n.t(`${this.path}.@@ErrorMessageForInvalidInput`, {
      args: { properties: invalidProperties.join(', ') },
    });
  }

  /** @inheritdoc */
  getTitleForError(): string {
    return this.i18n.t(`${this.path}.@@TitleForError`);
  }

  /** @inheritdoc */
  getTitleForInput(): string {
    return this.i18n.t(`${this.path}.@@TitleForInput`);
  }

  /** @inheritdoc */
  getTitleForQueryCode(): string {
    return this.i18n.t(`${this.path}.@@TitleForQueryCode`);
  }

  /** @inheritdoc */
  getTitleForResult(): string {
    return this.i18n.t(`${this.path}.@@TitleForResult`);
  }

  /** @inheritdoc */
  getTitleForStart(): string {
    return this.i18n.t(`${this.path}.@@TitleForStart`);
  }

  /** @inheritdoc */
  getTitleForSuccess(): string {
    return this.i18n.t(`${this.path}.@@TitleForSuccess`);
  }
}

File: server/makc2022--dotnet/Makc2022.Layer1/Query/QueryResource.cs
// Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using Microsoft.Extensions.Localization;

namespace Makc2022.Layer1.Query
{
    /// <summary>
    /// Ресурс запроса.
    /// </summary>
    public class QueryResource : IQueryResource
    {
        #region Properties

        private IStringLocalizer<QueryResource> Localizer { get; }

        #endregion Properties

        #region Constructors

        /// <summary>
        /// Конструктор.
        /// </summary>
        /// <param name="localizer">Локализатор.</param>
        public QueryResource(IStringLocalizer<QueryResource> localizer)
        {
            Localizer = localizer;
        }

        #endregion Constructors

        #region Public methods

        /// <inheritdoc/>
        public string GetErrorMessageForDefault()
        {
            return Localizer["@@ErrorMessageForDefault"];
        }

        /// <inheritdoc/>
        public string GetErrorMessageForInvalidInput(IEnumerable<string> invalidProperties)
        {
            return Localizer["@@ErrorMessageForInvalidInput", string.Join(", ", invalidProperties)];
        }

        /// <inheritdoc/>
        public string GetErrorMessageWithCode(string errorMessage, string code)
        {
            string title = Localizer["@@TitleForErrorCode"];

            return $"{errorMessage}. {title}: {code}".Replace("!.", "!").Replace("?.", "?");
        }

        /// <inheritdoc/>
        public string GetTitleForError()
        {
            return Localizer["@@TitleForError"];
        }

        /// <inheritdoc/>
        public string GetTitleForInput()
        {
            return Localizer["@@TitleForInput"];
        }

        /// <inheritdoc/>
        public string GetTitleForQueryCode()
        {
            return Localizer["@@TitleForQueryCode"];
        }    

        /// <inheritdoc/>
        public string GetTitleForResult()
        {
            return Localizer["@@TitleForResult"];
        }

        /// <inheritdoc/>
        public string GetTitleForStart()
        {
            return Localizer["@@TitleForStart"];
        }

        /// <inheritdoc/>
        public string GetTitleForSuccess()
        {
            return Localizer["@@TitleForSuccess"];
        }

        #endregion Public methods
    }
}
